### PR TITLE
ref: Restructure `snuba.clickhouse` module

### PR DESCRIPTION
**Pull request by tkaemming**
- Moves `snuba.clickhouse.ClickhousePool` to `snuba.clickhouse.native.ClickhousePool`.
- Moves `snuba.clickhouse.escape_col` to `snuba.utils.escape_col`.
- Moves column definitions from `snuba.clickhouse.*` to `snuba.clickhouse.columns.*`.
- Moves native reader from `snuba.reader.NativeDriverReader` to `snuba.clickhouse.native.NativeDriverReader`.
- Moves native writer from `snuba.writer.NativeDriverBatchWriter` to `snuba.clickhouse.native.NativeDriverBatchWriter`.
- Moves HTTP writer from `snuba.writer.HTTPBatchWriter` to `snuba.clickhouse.http.HTTPBatchWriter`.

### CODE REVIEW

File: snuba/util.py [hunk 1]
```diff
@@ -48,6 +49,21 @@ def string_col(dataset, col):
...
+    elif SAFE_COL_RE.match(col):
+        # Column is safe to use without wrapping.
+        return col
+    else:
+        # Column needs special characters escaped, and to be wrapped with
+        # backticks. If the column starts with a '-', keep that outside the
+        # backticks as it is not part of the column name, but used by the query
+        # generator to signify the sort order if we are sorting by this column.
+        col = ESCAPE_COL_RE.sub(r"\\\1", col)
+        return u'{}`{}`'.format(*NEGATE_RE.match(col).groups())
```
Comment (tkaemming) [inline snuba/util.py]: This is still relatively ClickHouse-specific, but not much more than several of the other processing bits in this file, I think.

File: snuba/util.py [hunk 2]
```diff
@@ -357,6 +375,8 @@ def raw_query(body, sql, client, timer, stats=None):
     Submit a raw SQL query to clickhouse and do some post-processing on it to
     fix some of the formatting issues in the result JSON
     """
+    from snuba.clickhouse.native import NativeDriverReader
```
Comment (tkaemming) [inline snuba/util.py]: 😬

**Comment by tkaemming** [comment]
I think that later it likely makes sense to move `snuba.clickhouse.columns` to `snuba.columns` and start working on splitting generic "snuba columns" from the concrete ClickHouse implementation.

**Review by fpacifici — COMMENTED** [review]
I would move raw_query out of util into a "query processor" module where you could fit the rest of the work you are doing related with query processing. Otherwise no objection.

**Review by fpacifici — APPROVED** [review]
Ok, if we need more discussion to find a place for raw_query

**Pull request merged**